src/Compilers/CSharp/Portable/Binder/Semantics/Operators/UnaryOperatorOverloadResolution.cs

```diff
@@ -263,16 +260,26 @@ private void GetAllBuiltInOperators(UnaryOperatorKind kind, BoundExpression oper
                 operators.Add(pointerOperator.Value);
             }
 
-            CandidateOperators(operators, operand, results, ref useSiteDiagnostics);
+            CandidateOperators(operators, operand, results, ref useSiteDiagnostics, skipNativeIntegerOperators: !operand.Type.IsNativeIntegerOrNullableNativeIntegerType());
```

Comment:
>CandidateOperators [](start = 12, length = 18)

Instead of filtering candidates from operators can we instruct GetSimpleBuiltInOperators to not put them there in the first place?